R/qualification-time-profile.R

```diff
@@ -78,144 +81,186 @@ plotQualificationTimeProfiles <- function(configurationPlan, settings) {
-      id = configurationPlanCurve$CurveOptions$LegendIndex,
-      secondAxis = curveOnSecondAxis,
-      # Use appropriate tlf atom for its default properties
-      plotFunction = tlf::addScatter
+  # Check necessity if dual axis by getting requested axes
+  requestedAxes <- sort(unique(c(simData$yAxis, obsData$yAxis)))
+  
+  # If only one axis requested, the only axis becomes the new y of axesProperties
+  if (isOfLength(requestedAxes, 1)) {
+    axesProperties$y <- axesProperties[[tolower(requestedAxes)]]
+    simData$yAxis <- ifNotNull(simData$yAxis, FALSE)
+    obsData$yAxis <- ifNotNull(obsData$yAxis, FALSE)
+    plotConfiguration <- updateQualificationTimeProfilePlotConfiguration(
+      simulatedMetaData = simMetaData,
+      observedMetaData = obsMetaData,
+      requestedAxes = requestedAxes,
+      axesProperties = axesProperties,
+      plotConfiguration = plotConfiguration
+    )
+    
+    plotObject <- tlf::plotTimeProfile(
+      data = simData,
+      observedData = obsData,
+      dataMapping = simDataMapping,
+      observedDataMapping = obsDataMapping,
+      plotConfiguration = plotConfiguration
     )
-    return(outputCurve)
+    return(plotObject)
   }
-  # Remove simulation name from path
-  outputPath <- ospsuite::toPathString(pathArray[-1])
-  # Get and convert output path values into display unit
-  simulationQuantity <- ospsuite::getQuantity(outputPath, simulation)
-  simulationPathResults <- ospsuite::getOutputValues(simulationResults, quantitiesOrPaths = outputPath)
-  molWeight <- simulation$molWeightFor(outputPath)
-
-  time <- ospsuite::toUnit(
-    "Time",
-    simulationPathResults$data[, "Time"],
-    axesProperties$x$unit
+  # If all Y, Y2 and Y3 axes are defined and used
+  # Keep Y and Y2 with a warning message
+  if (isOfLength(requestedAxes, 3)) {
+    logError(messages$warningTooManyAxes())
+    requestedAxes <- c("Y", "Y2")
+  }
+  if (isOfLength(requestedAxes, 1)) {
+    axesProperties$y <- axesProperties[[tolower(requestedAxes)]]
+    simData$yAxis <- ifNotNull(simData$yAxis, FALSE)
+    obsData$yAxis <- ifNotNull(obsData$yAxis, FALSE)
+    plotConfiguration <- updateQualificationTimeProfilePlotConfiguration(
+      simulatedMetaData = simMetaData,
+      observedMetaData = obsMetaData,
+      requestedAxes = requestedAxes,
+      axesProperties = axesProperties,
+      plotConfiguration = plotConfiguration
+    )
+    
+    plotObject <- tlf::plotTimeProfile(
+      data = simData,
+      observedData = obsData,
+      dataMapping = simDataMapping,
+      observedDataMapping = obsDataMapping,
+      plotConfiguration = plotConfiguration
+    )
+    plotObject <- updatePlotAxes(plotObject, axesProperties)
+    return(plotObject)
+  }
+  # Y, Y2 and Y3 defined and used
+  if (isOfLength(requestedAxes, 3)) {
+    logError(messages$warningTooManyAxes(
+      project = timeProfilePlan$Project,
+      simulation = timeProfilePlan$Simulation
+    ))
+    requestedAxes <- c("Y", "Y2")
```

Comment:
Why this check 2 times (here and above lines 189-191)?

Reply:
Copy/Paste typo.
I am going to remove it.